pkg/cloudprovider/drift.go

```diff
@@ -221,6 +204,65 @@ func (c *CloudProvider) isSubnetDrifted(ctx context.Context, nodeClaim *karpv1.N
 	return "", nil
 }
 
+// isKubeletIdentityDrifted returns drift if the kubelet identity has drifted
+func (c *CloudProvider) isKubeletIdentityDrifted(ctx context.Context, nodeClaim *karpv1.NodeClaim, _ *v1alpha2.AKSNodeClass) (cloudprovider.DriftReason, error) {
+	opts := options.FromContext(ctx)
+	logger := log.FromContext(ctx)
+
+	node, err := c.getNodeForDrift(ctx, nodeClaim)
+	if err != nil {
+		return "", err
+	}
+	if node == nil {
+		return "", nil
+	}
+
+	kubeletIdentityClientID := node.Labels[v1alpha2.AKSLabelKubeletIdentityClientID]
+	// The kubelet identity label is supposed to be set on every node, but prior to
+	// 1.4.0 it was not set by Karpenter. In order to avoid rolling all existing nodes,
+	// we don't count a missing kubelet identity as drift. This situation should resolve itself as
+	// image version and Kubernetes version drift is performed.
+	// TODO: This short-circuit should be removed post 1.4.0 (~2025-07-01)
```

Comment:
Discussed offline. Agreed, this approach works. Just need to come back to this in a later release.